src/org/labkey/targetedms/datasource/MsDataSourceUtil.java

```diff
@@ -203,9 +203,9 @@ else if(sourceTypes.size() > 1)
         {
             // We can get more than one source type by filename extension lookup. e.g. .raw extension is used both by Thermo and Waters
             // Try to resolve by looking up the instrument on which the data was acquired.
-            Instrument instrument = InstrumentManager.getInstrument(sampleFile.getInstrumentId());
-            if(instrument != null)
+            if(sampleFile.getInstrumentId() != null)
             {
+                Instrument instrument = InstrumentManager.getInstrument(sampleFile.getInstrumentId());
```

Comment:
This is probably fine because the foreign key should ensure there's a matching row in the Instrument table, but just wanted to confirm you intentionally removed the null check on instrument.

Reply:
Yes, if there is an instrument ID a matching row should exist in the Instrument table.